libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2127,6 +2130,65 @@ void GCS_MAVLINK::send_raw_imu()
 #endif
 }
 
+#if AP_MAVLINK_MSG_HIGHRES_IMU_ENABLED
+void GCS_MAVLINK::send_highres_imu()
+{
+    uint16_t fields_updated = 0;
+    Vector3f accel;
+    Vector3f gyro;
+#if AP_INERTIALSENSOR_ENABLED
+    const AP_InertialSensor &ins = AP::ins();
+    accel = ins.get_accel();
+    gyro = ins.get_gyro();
+    fields_updated |= 0x3F;
+#endif
+    
+    Vector3f mag;
+#if AP_COMPASS_ENABLED
+    const Compass &compass = AP::compass();
+    if (compass.get_count() >= 1) {
+        mag = compass.get_field() * 1000.0f; // convert to gauss
+    }
+    fields_updated |= 0xC0;
+#endif
+
+    float press_abs = 0.0f;
+    float temperature = 0.0f;
+    float altitude = 0.0f;
+    float diff_pressure = 0.0f;
+#if AP_BARO_ENABLED
+    const AP_Baro &barometer = AP::baro();
+    press_abs = barometer.get_pressure() * 0.01f;
+    temperature = barometer.get_temperature();
+    altitude = barometer.get_altitude();
+    diff_pressure = press_abs - barometer.get_ground_pressure() * 0.01f;
+    fields_updated |= 0xF00;
+#endif
+    // set to HIGHRES_IMU_UPDATED_ALL 
+    fields_updated = fields_updated == 0x0FFFF ? 0xFFFF : fields_updated;
```

Comment:
This has no effect the way you've written it. I think you meant to write:
```suggestion
    const uint16_t all_fields = (HIGHRES_IMU_UPDATED_XACC | <all of them...> | HIGHRES_IMU_UPDATED_TEMPERATURE
    fields_updated = (fields_updated == all_fields) ? HIGHRES_IMU_UPDATED_ALL : fields_updated;
```

Reply:
even better, instead of overwriting the variable just do this down in the  mavlink_msg_highres_imu_send() call